engine/engine_test.go

```diff
@@ -2901,6 +2923,131 @@ func TestEngineRecoversFromPanic(t *testing.T) {
 
 }
 
+func TestSparseHistogram(t *testing.T) {
+	opts := promql.EngineOpts{
+		Timeout:              1 * time.Hour,
+		MaxSamples:           1e10,
+		EnableNegativeOffset: true,
+		EnableAtModifier:     true,
+	}
+
+	cases := []struct {
+		name        string
+		count       int
+		metricsName string
+		labels      []string
+		query       string
+		mixedSample bool
+	}{
+		{
+			name:        "histogram_quantile(.80) with sparse histogram and old histograms",
```

Comment:
Let's replace all references to `sparse` histograms with `native` since that is the official term.

```suggestion
			name:        "histogram_quantile(.80) with native histogram and old histograms",
```